### DISCUSSION TITLE
Greenfields API: New "Limit" parameter for 'Get store on-chain wallet transactions' endpoint

### DISCUSSION THREAD

**Discussion by danielalexiuc** [btcpayserver/btcpayserver#2667] [Ideas / Feature Requests]
The "Get store on-chain wallet transactions" endpoint currently returns ALL transactions, with no way to limit the data returned.

/api/v1/stores/{storeId}/payment-methods/OnChain/{cryptoCode}/wallet/transactions

Please add an optional "limit" parameter (integer), so one can limit the amount of data coming back... for example, I might only be interested in the most recent 500 transactions.

**Comment by Kukks** [comment]
feature ACK, should also have a limit and take to be able to page the data

**Reply by woutersamaey** [reply]
For limit to work as expected, we also need to consider sorting. I imagine there is some sorting already going on and we just keep that? If not, we need to make sure sorting is applied so results are returned consistently.

Just making note of this.

If nobody is already doing this, I may take a look to implement in code.

**Reply by bolatovumar** [reply]
@woutersamaey there doesn't seem to be any sorting applied in this case. The only additional thing that can happen here is the filtering by `statusFilter` query string parameter.

**Reply by bolatovumar** [reply]
Draft PR here: https://github.com/btcpayserver/btcpayserver/pull/2688

**Comment by woutersamaey** [comment]
This discussion got me thinking of HATEOAS. Is this something we should leverage (maybe for Greenfield v2 at some point)?
